src/perf/isolates.md

```diff
@@ -0,0 +1,347 @@
+one required, unnamed argument.
+When the computation completes,
+it returns the callback's value back to the main isolate,
+and exits the spawned isolate.
+
+For example,
+consider this code that loads a large JSON blob from a file,
+and converts that JSON into custom Dart objects.
+If the json decoding process wasn't off loaded to a new isolate,
+this method would cause the UI to
+become unresponsive for several seconds.
+
+<?code-excerpt "lib/main.dart (isolate-run)"?>
+```dart
+// Produces a list of 211,640 photo objects.
+// (The JSON file is ~20MB.)
+Future<List<Photo>> getPhotos() async {
+  final String jsonString = await rootBundle.loadString('assets/photos.json');
+  final List<Photo> photos = await Isolate.run<List<Photo>>(() {
+    final List<Object?> photoData = jsonDecode(jsonString) as List<Object?>;
+    return photoData.cast<Map<String, Object?>>().map(Photo.fromJson).toList();
+  });
+  return photos;
+}
+```
+For a complete walkthrough of using Isolates to 
+parse JSON in the background, see [this cookbook recipe][].
+
+[this cookbook recipe]: {{site.url}}/cookbook/networking/background-parsing
+
+## Stateful, longer-lived isolates
+
+[`Isolate.run()`][] and `compute()` abstract a handful of lower-level, 
+isolate-related APIs to  simplify isolate management:
+
+- [`Isolate.spawn()`][] and [`Isolate.exit()`][]
+- [`ReceivePort`][] and [`SendPort`][]
+- [`send()`][] method
+
+When you use `Isolate.run` method,
+the new isolate immediately shuts down after it
+returns a single message to the main isolate.
+Sometimes, you'll need isolates that are long lived,
+and can pass multiple messages to each other over time.
+In Dart, you can accomplish this with the Isolate API
+and Ports.
+These long-lived isolates are colloquially known as _background workers_.
+
+Long-lived isolates are useful when you have a specific process that either
+needs to be run repeatedly throughout the lifetime of your application,
+or if you have a process that runs over a period of time
+and needs to yield multiple return values to the main isolate.
+
+### ReceivePorts and SendPorts
+
+Set up long-lived communication between isolates with two classes
+(in addition to Isolate):
+[`ReceivePort`][] and [`SendPort`][].
+These ports are the only way isolates can communicate with each other.
+
+`Ports` behave similarly to `Streams`,
+in which the `StreamController`
+or `Sink` is created in one isolate,
+and the listener is set up in the other isolate.
+In this analogy,
+the `StreamConroller` is called a `SendPort`,
+and you can "add" messages with the `send()` method.
+`ReceivePort`s are the listeners,
+and when these listeners receive a new message,
+they call a provided callback with the message as an argument.
+
+For an in-depth explanation on setting up two-way
+communication between the main isolate
+and a worker isolate,
+follow the examples in the [Dart documentation][].
+
+[Dart documentation]: {{site.dart-site}}/language/concurrency
```

Comment:
FWIW I have an easier way https://medium.com/@gaaclarke/isolate-agents-easy-isolates-for-flutter-6d75bf69a2e7.